validator/project_validator.go

```diff
@@ -657,6 +672,44 @@ func getAliasCoverage(p *model.Project, aliasMap map[string]model.ProjectAlias)
 	return aliasNeedsVariant, aliasNeedsTask, nil
 }
 
+// validateCheckRuns returns warnings if PR aliases are going to violate check run rules for the given config.
+func validateCheckRuns(p *model.Project, aliases model.ProjectAliases) ValidationErrors {
+	errs := ValidationErrors{}
+	aliasMap := map[string][]model.ProjectAlias{} // map of alias name to aliases
+	for _, a := range aliases {
+		if aliasMap[a.Alias] == nil {
+			aliasMap[a.Alias] = []model.ProjectAlias{}
+		}
+		aliasMap[a.Alias] = append(aliasMap[a.Alias], a)
+	}
+
+	tvPairs, err := p.BuildProjectTVPairsWithAlias(aliasMap[evergreen.GithubPRAlias], evergreen.GithubPRRequester)
+	if err != nil {
+		errs = append(errs, ValidationError{
+			Message: "problem getting task variant pairs for PR aliases",
+			Level:   Warning,
+		})
+		return errs
```

Comment:
Should we append to a list rather than return immediately here and on L700?

Reply:
I think there's no reason to continue if we aren't able to get the pairs here, since any following warnings would likely not actually be helpful. I'm not sure if we should keep going if we have trouble adding dependencies, but I think it's likely to be a similar situation if we were ever to run into an error there.